### PR TITLE
[stable/elasticsearch] add test

### PR DESCRIPTION
**Pull request by s12chung**
#### What this PR does / why we need it:

Add basic test to stable/elasticsearch. 

#### Which issue this PR fixes

No issue, adds new test.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by joejulian** [comment]
Added @cpanato since it seems like he's the most recently active on this chart.

**Comment by desaintmartin** [comment]
Hello,
Thanks! This looks interesting. Let's see how the CI handle this.
/ok-to-test
/assign

**Comment by desaintmartin** [comment]
It seems E2E test fails because of conflicts before even trying to test anything, could you solve them?

BTW, why do you call it "test framework" instead of "helm test"?

**Comment by s12chung** [comment]
@desaintmartin i rebased, it was conflicting the version names.

`test-framework` is a `initContainer` that copies the binary of the test framework, so that `{{ .Release.Name }}-test` container can use the binary and run the tests. code is based on: https://github.com/helm/charts/blob/2760b303449249b573a97c44e46b488e00c6db46/stable/mysql/templates/tests/test.yaml#L14, but there are other examples of `test-framework` (ex. https://github.com/helm/charts/blob/1eeb242811931dbe386f84401dc7117709ab8683/stable/mattermost-team-edition/templates/tests/mattermost-test.yaml#L14). trying to be consistent.

File: stable/elasticsearch/templates/tests/test-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
```
Comment (desaintmartin) [inline stable/elasticsearch/templates/tests/test-configmap.yaml:1]: I wonder if there is any way to deploy this it only when doing test...

**Comment by desaintmartin** [comment]
This is really nice and greatly improves quality of this chart.
/lgtm

**Comment by desaintmartin** [comment]
Could you rebase? It seems 1.23.0 is out.

File: stable/elasticsearch/templates/tests/test-configmap.yaml (hunk 1 above)
Comment (s12chung) [reply stable/elasticsearch/templates/tests/test-configmap.yaml:1]: this is how the other tests do it...

https://github.com/helm/charts/blob/bf877829d92dcb4bfc27163d60fcd403636915cc/stable/mysql/templates/tests/test-configmap.yaml#L1

https://github.com/helm/charts/blob/90e4ba91b3b0d933250bd3f5712a7199abb8703a/stable/consul/templates/test-config.yaml#L1

**Pull request merged**